pkg/fx/config/provider.go

```diff
@@ -0,0 +1,291 @@
+package config
+
+import (
+	crypto_ed25519 "crypto/ed25519"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/ipni/go-libipni/maurl"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/storacha/go-ucanto/client"
+	"github.com/storacha/go-ucanto/core/delegation"
+	"github.com/storacha/go-ucanto/did"
+	"github.com/storacha/go-ucanto/principal"
+	ed25519 "github.com/storacha/go-ucanto/principal/ed25519/signer"
+	ucanhttp "github.com/storacha/go-ucanto/transport/http"
+	"go.uber.org/fx"
+
+	"github.com/storacha/piri/pkg/config"
+	"github.com/storacha/piri/pkg/config/app"
+)
+
+var Module = fx.Module("config",
+	fx.Provide(
+		ProvideAppConfig,
+		// Provide named values for services that still need them
+		fx.Annotate(
+			ProvideUploadServiceConnection,
+			fx.ResultTags(`name:"upload_service_connection"`),
+		),
+	),
+)
+
+// ProvideAppConfig provides the new organized application configuration
+func ProvideAppConfig(cfg config.UCANServer) (app.AppConfig, error) {
```

Comment:
also, this method is huge. I'd create functions to create each config and assemble the AppConfig from the different bits here.